course_discovery/apps/core/utils.py

```diff
@@ -1,57 +1,89 @@
 import datetime
 import logging
+import re
+from collections import namedtuple
 
 from django.conf import settings
+from django_elasticsearch_dsl import Index
 
-from course_discovery.settings.process_synonyms import get_synonyms
-
+IndexMeta = namedtuple("IndexMeta", "name alias")
 logger = logging.getLogger(__name__)
 
+INDEX_ALIAS_REGEX = re.compile(r'^(\w+)(?=[_]\d{8}[_]\d{6})')
+INDEX_ALIAS_SLICE = slice(0, -16)
```

Comment:
Good point. I'll provide an example as comment lines.

Reply:
For the clean of the code, I put this slice into a constant with a clear name, so that bare numbers do not appear in the code and do not make anyone to think why exactly 0 and -16.